metrics/config.go

```diff
@@ -235,14 +241,22 @@ func createMetricsConfig(ops ExporterOptions, logger *zap.SugaredLogger) (*metri
 		if !allowCustomMetrics {
 			servingOrEventing := metricskey.KnativeRevisionMetrics.Union(
 				metricskey.KnativeTriggerMetrics).Union(metricskey.KnativeBrokerMetrics)
-			mc.recorder = func(ctx context.Context, ms stats.Measurement, ros ...stats.Options) error {
-				metricType := path.Join(mc.stackdriverMetricTypePrefix, ms.Measure().Name())
-
-				if servingOrEventing.Has(metricType) {
-					return stats.RecordWithOptions(ctx, append(ros, stats.WithMeasurements(ms))...)
+			mc.recorder = func(ctx context.Context, mss []stats.Measurement, ros ...stats.Options) error {
+				w := 0
```

Comment:
This is an efficient, but kind of hard to read block of code. 

I'm not going to say we shouldn't save the extra array allocation, but could we use more descriptive variables than "w" and "r"? I'm not sure what they're intended to stand for...

Reply:
`writeIndex` and `readIndex` :-)
I can make them more readable, this was more of a PoC:)